### PR TITLE
[TIC-82] Fix Cart for Mobile + Other UI fixes

### PR DESCRIPTION
**Pull request by henrykaus**
### Description
* Refactored `Cart` to use more semantic elements and reduce total number of nodes
* Refactored `CartItem` to be usable on mobile
* Fixed bug with logout button causing surrounding elements to be taller than ideal
* Refactored interactive elements at the top of `InstancesPage` to be responsive to mobile screens (and Prettier did some formatting)
* Changes "sm" breakpoints to "tab" in modals so that the mobile variants of modals appear
* Added `aria-labels` to interactable elements without otherwise accessible labeling
* Added `aria-hidden` to icons that don't need to be exposed to screen readers

**Cart/CartItem Before-After:**
image
image

**InstancesPage Before-After:**
image
image

**Example Mobile Modals Before-After:**
image
image

image
image

### Risks
There is the small potential that there are some places where the logout button is used and its spacing might be weird as a result of these changes.

### Validation
Visual

### Issue
[TIC-82](https://pdx-hkaus.atlassian.net/browse/TIC-82)

### Operating System
Windows 11

[TIC-82]: https://pdx-hkaus.atlassian.net/browse/TIC-82?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: client/src/components/Ticketing/ticketingmanager/showings/InstancesPage.tsx [hunk 1]
```diff
@@ -31,58 +34,88 @@ const InstancesPage = () => {
...
- * Based on active/inactive/all
- */
+  /**
+   * Based on active/inactive/all
+   *
+   * @param event
+   */
   const handleEventChange = (event) => {
     // TODO
   };
```
[the remark this answers is not in the document]
Comment (henrykaus) [reply client/src/components/Ticketing/ticketingmanager/showings/InstancesPage.tsx:48]: I'll add this to the TODO comment, but this function is to be filled in once we have active/inactive events working properly.

File: client/src/components/Authentication/login-button.tsx [hunk 2]
```diff
@@ -5,16 +5,9 @@ import {useAuth0} from '@auth0/auth0-react';
  *
  * @returns {React.ReactElement}
  */
-const LoginButton = () => {
+const LoginButton = (): React.ReactElement => {
```
[the remark this answers is not in the document]
Comment (henrykaus) [reply client/src/components/Authentication/login-button.tsx]: Fixed

File: client/src/components/Authentication/logout-button.tsx [hunk 3]
```diff
@@ -5,30 +5,31 @@ import {useAuth0} from '@auth0/auth0-react';
  *
  * @returns {React.ReactElement}
  */
-const LogoutButton = () => {
+const LogoutButton = (): React.ReactElement => {
```
[the remark this answers is not in the document]
Comment (henrykaus) [reply client/src/components/Authentication/logout-button.tsx]: Fixed

File: client/src/components/Ticketing/mainpage/footer.tsx [hunk 4]
```diff
@@ -40,7 +40,9 @@ const Footer = () => {
                              focus:border-indigo-600 focus:outline-none
                             "
                 id="exampleFormControlInput1"
-                placeholder="Email address"/>
+                placeholder="Email address"
+                aria-label='Email address'
```
[the remark this answers is not in the document]
Comment (henrykaus) [reply client/src/components/Ticketing/mainpage/footer.tsx]: Refactored the footer in general to be a lot simpler, but also made the text preceeding a label and removed the new aria-label

File: client/src/components/Ticketing/ticketingmanager/showings/InstancesPage.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (henrykaus) [reply client/src/components/Ticketing/ticketingmanager/showings/InstancesPage.tsx]: Updated the TODO

**Review by HaliTran — APPROVED** [review]
This looks great, thanks for making those changes!

**Pull request merged**